### PR TITLE
Feature/#103 add support for tuples

### PR DESCRIPTION
**Pull request by AGalabov**
* Added implementation of `ObjectSet` and a `uniq` function that utilizes it
* Added support for tuples through `array`

### CODE REVIEW

**Comment by AGalabov** [comment]
An interesting thing to consider is the [following comment](https://stackoverflow.com/a/57465199). It suggests that for OpenApi 3.1.0 we can use `prefixItems`. However the library `openapi3-ts` does not support it and neither does `openapi31-ts`. I am yet to debunk whether that is a valid OpenApi syntax. If it is I'd provide the needed type definitions to allow it manually (possibly open PRs to the respected libraries as well). Let me know your opinion on the matter

File: src/lib/object-set.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+      totalBuckets += 1;
+      totalValues += bucket.length;
+      if (bucket.length > 1) {
+        collisions += 1;
+      }
+    }
+    const hashEffectiveness = totalBuckets / totalValues;
+    return { totalBuckets, collisions, totalValues, hashEffectiveness };
+  }
+  private hashCodeOf(object: any): number {
```
Comment (georgyangelov) [inline src/lib/object-set.ts:75]: I'd put some blank lines here and there :D

File: src/openapi-generator.ts [hunk 2]
```diff
@@ -853,6 +854,35 @@ export class OpenAPIGenerator {
...
+          type: 'array',
+          items: uniqueSchemas[0],
+          minItems: tupleLength,
+          maxItems: tupleLength,
+        };
+      }
+
+      return {
+        ...this.mapNullableType('array', isNullable),
+        // TODO: Compact is extra
```
Comment (georgyangelov) [inline src/openapi-generator.ts:877]: Sorry, what?

**Review by georgyangelov — APPROVED** [review]

File: src/openapi-generator.ts (hunk 2 above)
Comment (AGalabov) [reply src/openapi-generator.ts:877]: There was a `compact` which was extra :smile: Now there is no compact ... but there was not TODO for the comment being extra. I'll remove it, thanks

**Comment by samchungy** [comment]
Btw looks like there's a better way to represent it in 3.1?

https://stackoverflow.com/questions/57464633/how-to-define-a-json-array-with-concrete-item-definition-for-every-index-i-e-a#57465199

**Comment by AGalabov** [comment]
@samchungy I've mentioned that exact Stack Overflow comment in my [initial comment](https://github.com/asteasolutions/zod-to-openapi/pull/107#issuecomment-1472159200). I didn't really manage to get a clarification that this is valid for OpenApi 3.1 - i.e with an online validator/documentation renderer. Please let me know if you find any.

**Comment by samchungy** [comment]
> @samchungy I've mentioned that exact Stack Overflow comment in my [initial comment](https://github.com/asteasolutions/zod-to-openapi/pull/107#issuecomment-1472159200). I didn't really manage to get a clarification that this is valid for OpenApi 3.1 - i.e with an online validator/documentation renderer. Please let me know if you find any.

I'll play around with something tomorrow at work and get back to you if it works 😄

**Pull request merged**